### PR TITLE
Improve `simpleExample`

### PR DESCRIPTION
**Pull request by Pamplemousse**
* cleaning: indentation, trailing spaces, random lines
  * add a small API example

### CODE REVIEW

File: src/org/zaproxy/zap/extension/simpleExample/SimpleExampleAPI.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
+/*
+/* Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2016 The ZAP Development Team
```
Comment (psiinon) [inline src/org/zaproxy/zap/extension/simpleExample/SimpleExampleAPI.java]: 2018?

File: src/org/zaproxy/zap/extension/simpleExample/SimpleExampleAPI.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import org.zaproxy.zap.extension.api.ApiException;
+import org.zaproxy.zap.extension.api.ApiImplementor;
+import org.zaproxy.zap.extension.api.ApiResponse;
+import org.zaproxy.zap.extension.api.ApiResponseElement;
+
+import net.sf.json.JSONObject;
+
+public class SimpleExampleAPI extends ApiImplementor {
+    // TODO shouldnt allow unsafe-inline styles - need to work out where they are being used
+    protected static final String CSP_POLICY =
```
Comment (psiinon) [inline src/org/zaproxy/zap/extension/simpleExample/SimpleExampleAPI.java]: Should remove this - its not used and will be very confusing for people :)

File: src/org/zaproxy/zap/extension/simpleExample/SimpleExampleAPI.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        this.addApiAction(new ApiAction(ACTION_HELLO_WORLD));
+    }
+
+    @Override
+    public String getPrefix() {
+        return PREFIX;
+    }
+
+    @Override
+    public String handleCallBack(HttpMessage msg) throws ApiException {
```
Comment (psiinon) [inline src/org/zaproxy/zap/extension/simpleExample/SimpleExampleAPI.java]: Remove as well

File: src/org/zaproxy/zap/extension/simpleExample/ExtensionSimpleExample.java [hunk 4]
```diff
@@ -45,147 +46,151 @@
...
+
+        extensionHook.addApiImplementor(this.api);
+
+        if (getView() != null) {
+            // Register our top menu item, as long as we're not running as a daemon
+            // Use one of the other methods to add to a different menu list
+            extensionHook.getHookMenu().addToolsMenuItem(getMenuExample());
+            // Register our popup menu item
+            extensionHook.getHookMenu().addPopupMenuItem(getPopupMsgMenuExample());
+            // Register a
```
Comment (psiinon) [inline src/org/zaproxy/zap/extension/simpleExample/ExtensionSimpleExample.java]: be nice to complete this comment :)

File: src/org/zaproxy/zap/extension/simpleExample/resources/Messages.properties [hunk 5]
```diff
@@ -2,11 +2,11 @@
...
-simpleExample.popup.msg				= Example right click menu invoked for:\n {0}
-simpleExample.popup.title			= Simple Example: Example right click menu
-simpleExample.topmenu.tools.msg		= Simple Example: Example message  
+simpleExample.desc								= A Simple Example Add-on
+simpleExample.error.nofile				= File not found : {0}
+simpleExample.panel.msg						= <html>This is a simple example of a status panel</html>
+simpleExample.panel.title					= Simple Example
+simpleExample.popup.msg						= Example right click menu invoked for:\n {0}
+simpleExample.popup.title					= Simple Example: Example right click menu
+simpleExample.topmenu.tools.msg		= Simple Example: Example message
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/simpleExample/resources/Messages.properties]: All the spacing before the equals signs could go away (IMHO), it's totally unnecessary.

(<sub>Ref: https://github.com/zaproxy/zaproxy/issues/276</sub>)

File: src/org/zaproxy/zap/extension/simpleExample/SimpleExampleAPI.java (hunk 3 above)
Comment (Pamplemousse) [reply src/org/zaproxy/zap/extension/simpleExample/SimpleExampleAPI.java]: I was kinda afraid to break things removing this one.
What is it used for?

File: src/org/zaproxy/zap/extension/simpleExample/ExtensionSimpleExample.java (hunk 4 above)
Comment (Pamplemousse) [reply src/org/zaproxy/zap/extension/simpleExample/ExtensionSimpleExample.java]: They are not really useful. 
Will just put `// As long as we are not running as a daemon` before the `if`.

File: src/org/zaproxy/zap/extension/simpleExample/ExtensionSimpleExample.java [hunk 6]
```diff
@@ -36,6 +36,7 @@
 import org.zaproxy.zap.utils.FontUtils;
 import org.zaproxy.zap.view.ZapMenuItem;
 
+
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/simpleExample/ExtensionSimpleExample.java]: It does not need another one.

File: src/org/zaproxy/zap/extension/simpleExample/ExtensionSimpleExample.java [hunk 7]
```diff
@@ -45,147 +46,151 @@
...
+    private ZapMenuItem menuExample;
+    private RightClickMsgMenu popupMsgMenuExample;
+    private AbstractPanel statusPanel;
 
-	private static final String EXAMPLE_FILE = "example/ExampleFile.txt";
-	
-	private ZapMenuItem menuExample;
-	private RightClickMsgMenu popupMsgMenuExample;
-	private AbstractPanel statusPanel;
+    private SimpleExampleAPI api = new SimpleExampleAPI(this);
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/simpleExample/ExtensionSimpleExample.java]: The initialisation can be delayed, to `init()` (or `hook`).

File: src/org/zaproxy/zap/extension/simpleExample/ExtensionSimpleExample.java [hunk 8]
```diff
@@ -45,147 +46,151 @@
...
-		return statusPanel;
-	}
-
-	private ZapMenuItem getMenuExample() {
+
+    @Override
+    public void hook(ExtensionHook extensionHook) {
+        super.hook(extensionHook);
+
+        extensionHook.addApiImplementor(this.api);
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/simpleExample/ExtensionSimpleExample.java]: The minimum ZAP version should be updated (to at least 2.6.0) in the add-on manifest (ZapAddOn.xml).

File: src/org/zaproxy/zap/extension/simpleExample/SimpleExampleAPI.java [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.simpleExample;
+
+import java.util.Locale;
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/simpleExample/SimpleExampleAPI.java]: This unused import should be removed.

File: src/org/zaproxy/zap/extension/simpleExample/SimpleExampleAPI.java [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.api.API;
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/simpleExample/SimpleExampleAPI.java]: Same for this one.

File: src/org/zaproxy/zap/extension/simpleExample/SimpleExampleAPI.java (hunk 3 above)
Comment (psiinon) [reply src/org/zaproxy/zap/extension/simpleExample/SimpleExampleAPI.java]: Thats the method that will be called if the add-on registers a callback with ZAP, ie a URL that is apparently on whatever target domain the add-on chooses :) In this case the add-on doesnt register for a callback, and I think its better it doesnt - thats too advanced for the intended use of this example and will only confuse people :)

File: src/org/zaproxy/zap/extension/simpleExample/ExtensionSimpleExample.java (hunk 7 above)
Comment (Pamplemousse) [reply src/org/zaproxy/zap/extension/simpleExample/ExtensionSimpleExample.java]: Shouldn't this be the same for the `LOGGER` variable as well then?

Comment (thc202) [reply src/org/zaproxy/zap/extension/simpleExample/ExtensionSimpleExample.java]: That one is final so it must be initialised earlier and being a class variable it depends on how/where it's being used (though in this example it's not heavily used). The extension might be disabled so initialising the API does not add much (just creates unnecessary garbage at some point).

**Comment by Pamplemousse** [comment]
@thc202 @kingthorin Any other comments or changes to be made?

File: src/org/zaproxy/zap/extension/simpleExample/SimpleExampleAPI.java [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.simpleExample;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.network.HttpMessage;
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/simpleExample/SimpleExampleAPI.java]: This import can be removed, not used.

**Comment by thc202** [comment]
Some of the commits could be fixed up.

**Comment by Pamplemousse** [comment]
What do you mean by "fixed up"?

**Comment by thc202** [comment]
The fixup step when doing interactive rebase, to merge some of the commits.

**Comment by Pamplemousse** [comment]
Rebased and squashed.

**Comment by Pamplemousse** [comment]
Aaah, the issue is from another repo, did not notice that. Thanks for noting.

Two last commit have been squashed as well :)

**Comment by thc202** [comment]
Thank you!

**Pull request merged**